aspnetcore/fundamentals/url-rewriting.md

```diff
@@ -0,0 +1,295 @@
+
+![Browser window with Developer Tools tracking the request and response](url-rewriting/_static/add_iis_url_rewrite.png)
+
+If you have an active IIS Rewrite Module with server-level rules configured that would impact your application in undesirable ways, you can disable the IIS Rewrite Module for an application with a change to your **web.config** file.
+```xml
+<configuration> 
+ <system.webServer> 
+  <modules> 
+   <remove name="RewriteModule" /> 
+  </modules> 
+ </system.webServer> 
+</configuration>
+```
+
+##### Unsupported features
+The middleware does not support the following IIS URL Rewrite Module features:
+* Global rules ([Basic Middleware #169](https://github.com/aspnet/BasicMiddleware/pull/169))
+* Rewrite Maps ([Basic Middleware #168](https://github.com/aspnet/BasicMiddleware/pull/168))
+* CustomResponse action ([Basic Middleware #135](https://github.com/aspnet/BasicMiddleware/issues/135))
+* Custom Server Variables ([Basic Middleware #183](https://github.com/aspnet/BasicMiddleware/issues/183))
+* trackAllCaptures ([Basic Middleware #178](https://github.com/aspnet/BasicMiddleware/pull/178))
+* Wildcards
+* LogRewrittenUrl
+
+##### Supported server variables
+The middleware supports the following IIS URL Rewrite Module server variables:
+* CONTENT_LENGTH
+* CONTENT_TYPE
+* HTTP_ACCEPT
+* HTTP_CONNECTION
+* HTTP_COOKIE
+* HTTP_HOST
+* HTTP_REFERER
+* HTTP_URL
+* HTTP_USER_AGENT
+* HTTPS
+* LOCAL_ADDR
+* QUERY_STRING
+* REMOTE_ADDR
+* REMOTE_PORT
+* REQUEST_FILENAME
+* REQUEST_URI
+
+> [!NOTE]
+> You can also obtain an `IFileProvider` via a `PhysicalFileProvider`. This approach may provide greater flexibility for the location of your rewrite rules files. Make sure that your rewrite rules files are deployed to the server at the path you provide.
+> ```csharp
+> PhysicalFileProvider fileProvider = new PhysicalFileProvider(Directory.GetCurrentDirectory());
+> ```
+
+#### Method-based rule
+Use `Add(Action\<RewriteContext\> applyRule)` to implement your own rule logic in a method. The `RewriteContext` exposes the `HttpContext` for use in your method. The `context.Result` determines how additional pipeline processing is handled.
+
+`context.Result` | Action
+--- | ---
+`RuleResult.ContinueRules` (default) | Continue applying rules
+`RuleResult.End Response` | Stop applying rules and send the response
+`RuleResult.SkipRemainingRules` | Stop applying rules and send the context to the next middleware
+
+[!code-csharp[Main](url-rewriting/sample/Startup.cs?name=snippet1&highlight=6)]
+
+The sample application demonstrates a method that redirects requests for paths that end with **.xml**. If you make a request for **/file.xml**, it's redirected to **/xmlfiles/file.xml**. The status code is set to 301 (Moved Permanently). For a redirect, you should explicitly set the status code of the response; otherwise, a 200 (OK) status code will be returned and the redirect might not occur on the client.
```

Comment:
I'll go with a "firmer" version then ...

> For a redirect, you must explicitly set the status code of the response; otherwise, a 200 (OK) status code will be returned and the redirect will not occur on the client.